beekeeper-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/generator/ExpiredHousekeepingMetadataGenerator.java

```diff
@@ -67,22 +69,35 @@ public ExpiredHousekeepingMetadataGenerator(String cleanupDelay) {
     this.clock = clock;
   }
 
+  // todo check what we do with the event list..if it's empty
   @Override
   public List<HousekeepingEntity> generate(ListenerEvent listenerEvent, String clientId) {
     List<HousekeepingEntity> housekeepingEntities = new ArrayList<>();
 
     switch (listenerEvent.getEventType()) {
     case CREATE_TABLE:
-      housekeepingEntities.add(generateHousekeepingEntity((CreateTableEvent) listenerEvent, clientId));
+      CreateTableEvent createTableEvent = (CreateTableEvent) listenerEvent;
```

Comment:
Maybe for the next time, but I feel this code could be refactored in some way. Maybe create an interface that extends all of this event types and add a function that is called isValid().

Then when you call each type you can just call the interface and will be valid for all types, you won't need to make an if for each specific event depending if it is table or partition.